### PR TITLE
sjawhar-legion-314: Pipeline visibility: enhanced legion status with phase grouping

### LINKED ISSUE #314: Pipeline visibility: enhanced `legion status` with phase grouping and worker state
## Context

Part of #277 (native coordination platform). Addresses pain point #1: "No pipeline visibility — no dashboard showing workers in each phase, what's blocked, what's idle."

Currently `legion status` shows a flat list of workers with ports. The data for full pipeline visibility already exists in `/state/collect` but isn't surfaced to the user.

## Goal

Enhance `legion status` (or add `legion dashboard`) to show a structured pipeline view: issues grouped by phase, worker status, blocked items, and idle issues.

## Acceptance Criteria

- [ ] `legion status <team>` output includes a "Pipeline" section that groups issues by their `status` field from `/state/collect` (Triage, Backlog, Todo, In Progress, Testing, Needs Review, Retro, Done)
- [ ] Each issue line shows: `<issue-id> <title (truncated to 50 chars)> [<worker-mode>:<worker-status>|IDLE|BLOCKED|NEEDS INPUT]`
- [ ] An issue is `BLOCKED` when `isBlocked === true` in collected state
- [ ] An issue is `NEEDS INPUT` when its labels include `user-input-needed`
- [ ] An issue is `IDLE` when it has no live worker and a non-skip suggested action
- [ ] Worker mode/status shown when `hasLiveWorker === true` (e.g., `[implement:running]`)
- [ ] A summary line at the end: `Total: N issues | Active: M workers | Blocked: B | Needs Input: I | Idle: J`
- [ ] `--json` flag outputs the same data as machine-readable JSON (for future dashboard integrations)
- [ ] When `/state/collect` fails or times out, the command prints an error message and exits with non-zero code (does not crash)
- [ ] Running `legion status <team>` with a test dataset of 10 issues across 5 phases produces correctly grouped output (verified by snapshot test)

## Testing Infrastructure

**Available:**
- `bun test` for unit tests
- Daemon HTTP API for integration testing
- Existing CLI command structure in `packages/daemon/src/cli/index.ts`

**Approach:**
- Unit test the formatting/grouping logic with mock `CollectedState` fixtures (snapshot tests for output format)
- Verify edge cases: 0 issues, all issues in one phase, all blocked, no live workers
- Integration test by running daemon with test data and verifying CLI output matches expected format

## Notes

- Follow existing CLI patterns in `packages/daemon/src/cli/index.ts` (citty `defineCommand`)
- The `/state/collect` response already contains all needed data (issue status, worker state, labels, PR state, CI status)

### PR DESCRIPTION
**Pull request by sjawhar**
Closes #314

## Summary

Enhance `legion status` to show a structured pipeline view grouping issues by their lifecycle phase (Triage → Done), with worker status indicators, blocked/idle detection, and a summary line.

### Changes

**State pipeline (`packages/daemon/src/state/`):**
- Add `title` field to `ParsedIssue`, `FetchedIssueData`, `IssueState`, `IssueStateDict` — carried through all layers (backends → enrichment → decision → serialization)
- GitHub backend extracts `title` from GraphQL project items (already fetched but previously discarded)
- Linear backend extracts `title` from raw issue data

**CLI (`packages/daemon/src/cli/`):**
- New `pipeline.ts` module with formatting/grouping logic, decoupled from CLI for testability
- Enhanced `cmdStatus` calls `/state/fetch-and-collect` to get pipeline data
- `--json` flag outputs machine-readable JSON (for future dashboard integrations)
- Error handling: graceful failure when daemon unreachable or `/state/collect` fails

**Tests (`packages/daemon/src/cli/__tests__/pipeline.test.ts`):**
- 10-issue fixture across 5 phases for snapshot testing
- Edge cases: empty state, all-in-one-phase, all-blocked, no-workers, title truncation
- Status indicator priority tests (BLOCKED > NEEDS INPUT > worker:status > IDLE)

### Output format

```
Legion Status: sjawhar/2
========================================
Daemon: RUNNING (port 13370)
Workers: 3

Pipeline
----------------------------------------

Backlog (2)
  sjawhar-legion-101 Set up authentication service with JWT tok… [IDLE]
  sjawhar-legion-102 Add rate limiting to API endpoints [BLOCKED]

In Progress (2)
  sjawhar-legion-104 Implement webhook delivery system with ret… [implement:running]
  sjawhar-legion-105 Fix memory leak in event processor causing… [NEEDS INPUT]

...

Total: 10 issues | Active: 3 workers | Blocked: 1 | Needs Input: 1 | Idle: 4
```

### CODE REVIEW

File: packages/daemon/src/daemon/server.ts [hunk 1]
```diff
@@ -739,7 +739,7 @@ export function startServer(opts: ServerOptions): {
           workers.delete(workerId);
           crashHistory.delete(workerId);
           await persistState();
-          unsubscribeWorkerFromEnvoy(entry.sessionId);
+          unsubscribeAllWorkerTopics(entry.sessionId);
```
Comment (sjawhar) [inline packages/daemon/src/daemon/server.ts]: **[P3]:** Tangential cleanup — these two `unsubscribeWorkerFromEnvoy` → `unsubscribeAllWorkerTopics` renames fix stale references but are unrelated to the pipeline feature. Harmless, just noting for PR scope clarity.

**Comment by sjawhar** [comment]
Superseded by V2 design (see #394).
